src/tests/nativeaot/CustomMain/CustomMain.csproj

```diff
@@ -5,7 +5,8 @@
     <StaticLibraryPrefix Condition="'$(TargetOS)' != 'windows'">lib</StaticLibraryPrefix>
 
     <!-- TODO-LLVM: Remove when https://github.com/WebAssembly/wasi-sdk/issues/367 is resolved -->
-    <EmccExtraArgs Condition="'$(TargetOS)' == 'wasi'">-Xlinker -export-if-defined=__main_argc_argv</EmccExtraArgs>
+    <EmccExtraArgs Condition="'$(TargetOS)' == 'wasi'">-Wl,--export-if-defined=__main_argc_argv</EmccExtraArgs>
+    <CLRTestExitCode>0</CLRTestExitCode>
```

Comment:
`<CLRTestExitCode>0</CLRTestExitCode>` should be hoisted to `src/tests/nativeaot/Directory.Build.props`.

Reply:
I recall doing it this way because I only wanted to modify the tests that were relevant to WASI.  For example, `src/tests/nativeaot/SmokeTests/HardwareIntrinsics/Program.cs` still uses a status code of 100 to report success, which is okay because WASI doesn't run that test anyway.  Should I modify all the tests under `src/tests/nativeaot` to use 0 instead of 100?

Reply:
Either that, or you can override the `</CLRTestExitCode>` back to `100` in the those tests.

Reply:
Hmm, it looks like there are still quite a few tests returning 100 for success under `src/tests/nativeaot`, so making the default 0 and overriding it will make the diff roughly as big as leaving the default 100 and overriding that.  Given that @pavelsavara is doing the "write the exit code to stdio" trick upstream, perhaps this is all irrelevant anyway?

Reply:
> the default 0 and overriding it will make the diff roughly as big as leaving the default 100 and overriding that

Let's leave it as is then.

> perhaps this is all irrelevant anyway?

Note the trick only applies to upstream hosting models. With your WASI change, I hope we'll get to use it before we need to run the whole runtime test suite :).

Reply:
https://github.com/WebAssembly/wasi-cli/pull/44